lib/Dialect/FIRRTL/Transforms/IMDeadCodeElim.cpp

```diff
@@ -42,6 +44,16 @@ static bool isDeletableDeclaration(Operation *op) {
   return !hasDontTouch(op);
 }
 
+/// Return true if the annotation is ok to drop when the target is dead.
+static bool isWeakReferencingAnnotation(Annotation anno) {
+  if (!anno.isClass(omirTrackerAnnoClass))
+    return false;
+
+  auto tpe = anno.getMember<StringAttr>("type");
```

Comment:
This is NOT acceptable.  No pass should be interpreting annotations except passes explicitly dealing with those annotations.